src/PRJobProducer/Program.cs

```diff
@@ -29,94 +33,110 @@ public class Program
-                        Console.WriteLine("--user was provided with no --token.");
+                        Console.WriteLine("--github-user was provided with no --github-token.");
                         return -1;
                     }
 
-                    botLoginName = optionBotUser.Value();
-                    client = GetClientForUser(botLoginName, optionBotToken.Value());
+                    botLoginName = githubUser.Value();
+                    client = GetClientForUser(botLoginName, githubUserToken.Value());
                 }
-                else if (optionAppId.HasValue())
+                else if (githubAppId.HasValue())
                 {
-                    if (!optionAppKeyPath.HasValue())
+                    if (!githubAppKeyPath.HasValue())
                     {
-                        Console.WriteLine("--app-id was provided with no --key-file.");
+                        Console.WriteLine("--github-app-id was provided with no --github-app-key-file.");
                         return -1;
                     }
 
-                    if (!optionAppInstallationId.HasValue())
+                    if (!githubAppInstallationId.HasValue())
                     {
-                        Console.WriteLine("--app-id was provided with no --install-id.");
+                        Console.WriteLine("--github-app-id was provided with no --github-app-install-id.");
                         return -1;
                     }
 
-                    if (!long.TryParse(optionAppInstallationId.Value(), out long installId))
+                    if (!long.TryParse(githubAppInstallationId.Value(), out long installId))
                     {
-                        Console.WriteLine("--install-id is not a valid long.");
+                        Console.WriteLine("--github-app-install-id is not a valid long.");
                     }
 
                     botLoginName = AppName + "[bot]";
-                    client = await GetClientForApp(optionAppId.Value(), optionAppKeyPath.Value(), installId);
+                    client = await GetClientForApp(githubAppId.Value(), githubAppKeyPath.Value(), installId);
                 }
                 else
                 {
-                    Console.WriteLine("Cannot authenticate with GitHub. Neither a --user nor an --app-id has been provided.");
+                    Console.WriteLine("Cannot authenticate with GitHub. Neither a --github-user nor an --github-app-id has been provided.");
                     return -1;
                 }
 
+
+                if (jobsPath.HasValue())
+                {
+                    JobFileSystem = new LocalFileSystem(jobsPath.Value());
+                }
+                else if (azureStorageConnectionString.HasValue())
+                {
+                    if (!azureStorageFileShareName.HasValue())
+                    {
+                        Console.WriteLine("--azure-storage-connection-string was provided with no --azure-storage-file-share.");
```

Comment:
nit: this check could be more robust by checking the existence of both instead of one and then the other.

Reply:
I added an else to cover this case.